### PR TITLE
updated POM with MQTT libs (and fixed indentation issues)

### PR DESCRIPTION
**Pull request by jenseliasson**
Added  Paho lib, fixed minor issues

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -82,15 +82,21 @@
...
 		<dependency>
-      <groupId>org.springframework</groupId>
-      <artifactId>spring-websocket</artifactId>
+			<groupId>org.springframework</groupId>
+			<artifactId>spring-websocket</artifactId>
+		</dependency>
+
+		<dependency>
+			<groupId>org.eclipse.paho</groupId>
+			<artifactId>org.eclipse.paho.client.mqttv3</artifactId>
```
Comment (borditamas) [inline pom.xml:120]: Do you plan to implement some methods into ArrowheadService.calss for mqtt?

Comment (jenseliasson) [reply pom.xml:120]: That's the plan. I will implement broker management and methods for interacting with the core systems.

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 2]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+	 */
+	public MqttClient connectMQTTBroker(final MqttCallback handler, final String brokerAddress, final int port, final String caFile,
+							final String certFile, final String keyFile, final String password, final String clientId) throws Exception {
+		if (handler == null) {
+			throw new InvalidParameterException("handler cannot be null.");
+		}
+		if (Utilities.isEmpty(brokerAddress)) {
+			throw new InvalidParameterException("brokerAddress cannot be null or blank.");
+		}
+		if (port <= 0) {
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: Do the same like in WS connection:
```
if (port < CommonConstants.SYSTEM_PORT_RANGE_MIN || port > CommonConstants.SYSTEM_PORT_RANGE_MAX){
           throw new InvalidParameterException("Port must be between " + CommonConstants.SYSTEM_PORT_RANGE_MIN + " and " + CommonConstants.SYSTEM_PORT_RANGE_MAX + ".");
}
```

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 3]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+
+		final MemoryPersistence persistence = new MemoryPersistence();
+		final MqttClient client = new MqttClient(brokerAddress, clientId, persistence);
+		final MqttConnectOptions connOpts = new MqttConnectOptions();
+
+		if ( !Utilities.isEmpty(caFile) && !Utilities.isEmpty(certFile) && !Utilities.isEmpty(keyFile) ) {
+			try {
+				//connOpts.setSocketFactory(SslUtil.getSocketFactory(caFile, certFile, keyFile, password));
+			} catch(Exception err) {
+				System.out.println("MQTTS security exception: " + err);
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: use `logger.error(...)`

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 4]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+		}
+
+		final MemoryPersistence persistence = new MemoryPersistence();
+		final MqttClient client = new MqttClient(brokerAddress, clientId, persistence);
+		final MqttConnectOptions connOpts = new MqttConnectOptions();
+
+		if ( !Utilities.isEmpty(caFile) && !Utilities.isEmpty(certFile) && !Utilities.isEmpty(keyFile) ) {
+			try {
+				//connOpts.setSocketFactory(SslUtil.getSocketFactory(caFile, certFile, keyFile, password));
+			} catch(Exception err) {
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java:643]: missing final

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 5]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+		}
+		if (Utilities.isEmpty(clientId)) {
+			throw new InvalidParameterException("clientId cannot be null or blank.");
+		}
+
+		final MemoryPersistence persistence = new MemoryPersistence();
+		final MqttClient client = new MqttClient(brokerAddress, clientId, persistence);
+		final MqttConnectOptions connOpts = new MqttConnectOptions();
+
+		if ( !Utilities.isEmpty(caFile) && !Utilities.isEmpty(certFile) && !Utilities.isEmpty(keyFile) ) {
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: Would be more user friendly if caFile, certFile, keyFile would be retrieved from SSLProperties component.
see: `private KeyStore getKeyStore()` and `private KeyStore getTrustStore()`

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 6]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+			try {
+				//connOpts.setSocketFactory(SslUtil.getSocketFactory(caFile, certFile, keyFile, password));
+			} catch(Exception err) {
+				System.out.println("MQTTS security exception: " + err);
+				System.exit(-1);
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: What is the reason of system exit? I suggest just to throw `ArrowheadException` and let the developer decide what should happen.

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 7]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+		}
+		connOpts.setCleanSession(true);
+		client.setCallback(handler);
+		
+		logger.debug("Connecting to MQTT(S) broker: " + brokerAddress);
+		client.connect(connOpts);
+
+		return client;
+	}
+
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java:656]: I think having a `public void closeMQTTBrokerConnection(....)` would be useful.

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 8]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
 		return false;
 	}
 
+	//-------------------------------------------------------------------------------------------------
+	/**
+	 * Connect to MQTT broker.
+	 *
+	 * @param handler string id of the WebSocketcallback handler
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: it is an MqttCallback handler

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 9]
```diff
@@ -584,6 +594,54 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+	 * @param brokerAddress address to the broker
+	 * @param port port to the broker
+	 * @param clientId the client name to use
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java:601]: `password` is missing
`caFile`, `certFile`, `keyFile` is also missing but these coud be retrieved from `SSLProperties` component

**Review by borditamas — CHANGES_REQUESTED** [review]

**Comment by jenseliasson** [comment]
Code is updated

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 10]
```diff
@@ -584,6 +591,99 @@ public boolean closeWSConnection(final String wsManagerId) {
...
+	 * @throws ArrowheadException if a certificate error occurs
+	 */
+	public MqttClient connectMQTTBroker(final MqttCallback handler, final String brokerAddress, final int port, final String clientId, final String password) throws Exception {
+		if (handler == null) {
+			throw new InvalidParameterException("handler cannot be null.");
+		}
+		if (Utilities.isEmpty(brokerAddress)) {
+			throw new InvalidParameterException("brokerAddress cannot be null or blank.");
+		}
+		if (port <= 0) {
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: remove this

**Review by borditamas — APPROVED** [review]

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java [hunk 11]
```diff
@@ -625,16 +626,20 @@ public MqttClient connectMQTTBroker(final MqttCallback handler, final String bro
...
+			connOpts.setUserName(mqttBrokerUsername);
+			connOpts.setPassword(mqttBrokerPassword.toCharArray());
+		}
+
 		if(sslProperties.isSslEnabled()) {
 			try {
 				Properties sslMQTTProperties = new Properties();
-				final KeyStore keyStore = getKeyStore();
 				sslMQTTProperties.put(SSLSocketFactoryFactory.KEYSTORE, keyStore);
-				sslMQTTProperties.put(SSLSocketFactoryFactory.KEYSTOREPWD, sslProperties.getKeyStorePassword());
```
Comment (borditamas) [inline src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: What was wrong with using SSLProperties component?

File: src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java (hunk 10 above)
Comment (jenseliasson) [reply src/main/java/ai/aitia/arrowhead/application/library/ArrowheadService.java]: removed

**Comment by jenseliasson** [comment]
Please  merge this pull request

**Pull request merged**